### PR TITLE
fix(analyze): support signatures based on signatures

### LINKED ISSUE #4009: Analyze mode doesn't support sigs-on-sigs
## Description

The current tracee analyze mode is not well maintained or tested, so the sigs-on-sigs option is not yet supported yet.
For compatibility we should support it for now until the full analyze mode will be implemented.

## Output of `tracee version`:

```
 v0.20.0-93-ge8d7a3516
```

### PR DESCRIPTION
**Pull request by AlonZivony**
### 1. Explain what the PR does

Support signatures based on signatures in the analyze mode.
This is done in the purpose of making all signatures that are compatible with tracee to be also compatible with the analyze mode.

Fix #4009

### CODE REVIEW

File: cmd/tracee/cmd/analyze.go [hunk 1]
```diff
@@ -240,7 +240,7 @@ func getSigsNames(sigs []detect.Signature) []string {
 			logger.Warnw("Failed to get signature metadata", "err", err)
 			continue
 		}
-		sigsNames = append(sigsNames, sigMeta.Name)
+		sigsNames = append(sigsNames, sigMeta.ID)
```
Comment (geyslan) [inline cmd/tracee/cmd/analyze.go:310]: I think this change would require changing getSigsNames to getSigsIDs, wouldn't?

**Review by geyslan — COMMENTED** [review]
On a cursory review, it looks ok, even though I haven't tested it yet.

How about moving the pipeline logic into a new file like `analyze_pipeline.go`?

File: cmd/tracee/cmd/analyze.go (hunk 1 above)
Comment (AlonZivony) [reply cmd/tracee/cmd/analyze.go:310]: You are right👌

**Comment by geyslan** [comment]
Hey @AlonZivony I think this is good for merging right? Could you just rebase again?

File: pkg/analyze/pipeline.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+			case event := <-sigsEvents:
+				printInput <- event
+				engineInput <- event
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
+	go sigEngine.Start(ctx)
```
Comment (geyslan) [inline pkg/analyze/pipeline.go:72]: Is missing a wg.Add(1)?

File: pkg/analyze/pipeline.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+// printEvents starts a goroutine that prints the received events if they are chosen.
+// It is similar to the Sink stage in the normal pipeline.
+func printEvents(ctx context.Context, wg *sync.WaitGroup, eventsToPrint []string, eventsInput <-chan protocol.Event) {
+	wg.Add(1)
+
+	eventsToPrintMap := make(map[string]struct{}, len(eventsToPrint))
+	for _, eventName := range eventsToPrint {
+		eventsToPrintMap[eventName] = struct{}{}
+	}
+	go func() {
```
Comment (geyslan) [inline pkg/analyze/pipeline.go:162]: nit: move the (156 line) wg.Add(1) to here.

**Comment by AlonZivony** [comment]
I close this PR as we plan to merge the full analyze mode, which will fix the issue.

**Pull request closed without merging**